src/PAModel/Utility/Utilities.cs

```diff
@@ -173,8 +173,20 @@ public static string GetRelativePath(string full, string basePath)
         private static string AppendDirectorySeparatorChar(string path)
         {
             // Append a slash only if the path is a directory and does not have a slash.
-            if (!Path.HasExtension(path) &&
-                !path.EndsWith(Path.DirectorySeparatorChar.ToString()))
+            var isFilePath = File.Exists(path);
+            var isDirectoryPath = Directory.Exists(path);
```

Comment:
This should be a "pure" string operation, we shouldn't depend on the file system.  (I realized this after looking at the tests). 
We'd like to just call Path.GetRelativePath() - that correctly handles the inputs here -  but it's not available in this framework, so we handrolled it ourselves. Interesting to see where they added it:   https://github.com/dotnet/corefx/pull/11689/files

Their impl is significantly more complex than the impl here, which suggests that there are other corner cases we're missing.

Reply:
That's certainly ideal ;)  Based on the `Path.GetRelativePath` implementation it states that the `relativeTo` parameter is ALWAYS considered to be a directory.  If we also make that the premise for the custom implementation here then the problem becomes much simpler.  It would also make transitioning to the NetCore version down the road pretty seamless.  Thoughts?

Reply:
The current impl is at:  https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L861

That implementation is pretty complex (lots of helpers, platform-specific code, unsafe pointers), so it looks like we can't just port it. 

The 'relativeTo' parameter is (basePath) is always a directory.

Reply:
Actually, 1st arg is always file, 2nd arg is always dir, so that should simplify this. 
Here's a proposal: https://github.com/microsoft/PowerApps-Language-Tooling/pull/204 

How's that look? (github won't let me add you as a reviewer).

Some of the tests were accidentally commented out, so I reenabled those. 
Plus one test was bogus.